### ISSUE TITLE
make feature_config FAMILY=nrf51  causes `fatal error: feature_config.template.h: No such file or directory `

### ISSUE THREAD

**Issue by aguaviva** [dakhnod/BLEnky#21]
This is my full output:

```
$ make feature_config FAMILY=NRF51
ERROR: ld.so: object '/usr/lib/x86_64-linux-gnu/libstdc++.so.6' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
java -jar ../CMSIS_Configuration_Wizard.jar src/config/feature_config.template.h
Unable to open src/config/feature_config.template.h
```

**Comment by aguaviva** [comment]
- Alright, it looks I need to use `CMSIS_Configuration_Wizard.jar`, hopefully this is a good one? https://github.com/MZachmann/CMSIS_Configuration/blob/master/dist/CMSIS_Configuration_Wizard.jar

- Also, I'd be helpful to  mentions we need to clone with submodules: `git clone --recurse-submodules git@github.com:dakhnod/BLEnky.git`

**Comment by aguaviva** [comment]
I needed to install jinja like this
```
 pip3 install Jinja2
```

then I had to create a python script in `/src/config/process.py` to generate `src/config/feature_config.template.h`
``` python
from jinja2 import Environment, FileSystemLoader

env = Environment(loader = FileSystemLoader('.'))

template = env.get_template('feature_config.template.h.jinja')

with open("feature_config.template.h", 'w') as f:
    print(template.render(), file = f)
``` 

and only then `make FAMILY=NRF51 feature_config` sort of worked, because it threw errors when opening some branches of the tree,  so I didn't save anything, and only then I could compile things... hopefully this helps other people.

**Comment by dakhnod** [comment]
Yep, sorry for the confusion. I need to auto-create feature_config.template.h...

S130 should work without problems, even on nRF51822 qfaa. S130 is specificaly and exclusively made for nrf51822...

**Comment by dakhnod** [comment]
Also, jinja can be parsed here: http://jinja.quantprogramming.com/

**Comment by dakhnod** [comment]
Thanks for bringing this up, the template should be parsed automatically now.

In any case, BLEnky was designed to work with s130 and does so on all my nrf51822 boards, even the smallest ones.
